### DISCUSSION TITLE
Pulling response headers from request

### DISCUSSION THREAD

**Discussion by meain** [microsoftgraph/msgraph-sdk-go#330] [Q&A] [answered]
How would I be able to pull response headers like `Retry-After` mentioned in [docs](https://learn.microsoft.com/en-us/graph/throttling#best-practices-to-handle-throttling) or info in them when being throttled?

``` go
// just for context
item, err := client.DrivesById(driveID).ItemsById(itemID).Get(context.Background(), nil)
```
I might be missing something, but looking at the err(`odataerrors.ODataError`), while it has info that it was throttled  does not have anything more than that.

**Comment by baywet** [comment, marked as answer]
Hi @meain 
The SDK already has a [retry-handler](https://github.com/microsoft/kiota-http-go/blob/main/retry_handler.go) that automatically reads the headers and retries failed responses across all the API surface for you.
You can either create your own handler and replace this one in the client configuration if you'd like to change the behavior across all requests.
Or you can use the [RequestHandlerOption](https://github.com/microsoft/kiota-abstractions-go/blob/main/request_handler_option.go) to intercept the response handlers for a one of request.
